cmd/crd-linter/linters/interface.go

```diff
@@ -0,0 +1,32 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package linters
+
+import v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+
+// A Linter evaluates a CRD resource and provides information about policy violations.
+type Linter interface {
+	// Name returns the name of this linter, used during report output
+	Name() string
+
+	// Execute runs the linter against the given custom resource definition
+	Execute(*v1.CustomResourceDefinition) []string
```

Comment:
Have you considered a more "structured" return type for this function? When implementing new future linters, I would prefer a "tighter" API than just `[]string`. To understand a lint violation, I think the user at least need some reference to the violation inside the CRD-file. Path and/or line number?

Reply:
Done ✅